jina/peapods/runtimes/zmq/zed.py

```diff
@@ -361,7 +362,14 @@ def expect_parts(self) -> int:
 
         :return: expected number of partial messages
         """
-        return self.args.num_part if self.message.is_data_request else 1
+        if self.message.is_data_request:
+            if self.args.socket_in == SocketType.ROUTER_BIND:
```

Comment:
I think this logic should be based on the presence or not of `routing_graph` and not on the type of socket

Reply:
Haha, on this is more subtle. You always have a routing graph. But only the the HeadPeas should respect the `expected_parts` coming from the routing graph. TailPeas after sharding for example still need to check `self.args.num_part`.

I could at an argument to HeadPeas, similar to `dynamic_out_routing` to be `dynamic_in_routing`. Would you prefer that?

Reply:
I would do, I think is less error-prone and more explicit. 

I can imagine int he future for some reason setting a `ROUTER_BIND` in a very different context and having a very bad surprise